boomerang/src/nicp_to_icp.rs

```diff
@@ -97,32 +102,32 @@ pub async fn try_retrieve_icp(target: Principal) -> Result<Nat, BoomerangError>
     let boomerang_id = self_canister_id();
     let subaccount = derive_subaccount_unstaking(target);
 
-    let target_account = Account {
+    let boomerang_account = Account {
         owner: boomerang_id,
         subaccount: Some(subaccount),
     };
 
-    let icp_balance_e8s: u64 = match icp_client.balance_of(target_account).await {
+    let icp_balance_e8s: u64 = match icp_client.balance_of(boomerang_account).await {
         Ok(balance) => balance.0.try_into().unwrap(),
         Err((code, message)) => {
-            return Err(BoomerangError::GenericError { code, message });
+            return Err(BoomerangError::BalanceOfError(format!(
+                "code: {code} - message: {message}"
+            )));
         }
     };
 
-    if icp_balance_e8s < TRANSFER_FEE {
-        return Err(BoomerangError::NotEnoughICP);
+    if icp_balance_e8s == 0 {
```

Comment:
Why?

Reply:
It is because of the merge of the past repo to waterNeuron. I am changing it back to:
```
if icp_balance_e8s < TRANSFER_FEE {
        return Err(BoomerangError::NotEnoughICP);
        ```